engine/src/test/java/io/camunda/zeebe/engine/state/deployment/DecisionStateTest.java

```diff
@@ -258,6 +258,62 @@ void shouldFindDeployedDecisionRequirementsByKey() {
         .isEqualTo(drg2.getDecisionRequirementsKey());
   }
 
+  @DisplayName("should return empty if no decision found for DRG key")
+  @Test
+  void shouldReturnEmptyIfNoDecisionFoundForDrgKey() {
+    // given
+    final var unknownDrgKey = 1L;
+
+    // when
+    final var decisions = decisionState.findDecisionsByDecisionRequirementsKey(unknownDrgKey);
+
+    // then
+    assertThat(decisions).isEmpty();
+  }
+
+  @DisplayName("should find decisions by DRG key")
+  @Test
+  void shouldFindDecisionsByDrgKey() {
+    // given
+    final var drg1 = sampleDecisionRequirementsRecord().setDecisionRequirementsKey(10L);
+    final var drg2 = sampleDecisionRequirementsRecord().setDecisionRequirementsKey(20L);
+
+    final var decision1 =
+        sampleDecisionRecord()
+            .setDecisionKey(1L)
+            .setDecisionRequirementsKey(drg1.getDecisionRequirementsKey());
+    final var decision2 =
+        sampleDecisionRecord()
+            .setDecisionKey(2L)
+            .setDecisionRequirementsKey(drg1.getDecisionRequirementsKey());
+    final var decision3 =
+        sampleDecisionRecord()
+            .setDecisionKey(3L)
+            .setDecisionRequirementsKey(drg2.getDecisionRequirementsKey());
```

Comment:
❓ What happens if 2 drgs refer to the same decision? We'd create a new version of that decision right? Would that new version have the same key, or a different key?

Reply:
The keys are unique. The new version of the decision has a different key.